### PR TITLE
refactor: reuse edge runtime helpers

### PR DESCRIPTION
**Pull request by Kikobeats**
- `@edge-runtime/cookies` exports `splitCookiesString` so it can be used rather than re-implement the same function again.
- `@edge-runtime/node-utils` exports `toOutgoingHeaders` that is equivalent to Next.js `toNodeOutgoingHttpHeaders` function.

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 62170f27301654cc440e3ed468426ce3e9a11143

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 62170f27301654cc440e3ed468426ce3e9a11143

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by huozhi — DISMISSED** [review]

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 8f03752493813a19a6bbbbec6b957d5bb9c1bd49

`pnpm testheadless packages/next/src/server/lib/mock-request.test.ts` 
- MockedResponse > should handle set-cookie headers correctly

<details>
<summary>Expand output</summary>

  ● MockedResponse › should handle set-cookie headers correctly

    expect(received).toEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 0

      Object {
        "set-cookie": Array [
    -     "foo=bar",
          "bar=foo",
        ],
      }

      74 |
      75 |     res.setHeader('set-cookie', ['foo=bar', 'bar=foo'])
    > 76 |     expect(res.getHeaders()).toEqual({
         |                              ^
      77 |       'set-cookie': ['foo=bar', 'bar=foo'],
      78 |     })
      79 |

      at Object.<anonymous> (../packages/next/src/server/lib/mock-request.test.ts:76:30)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -5195,13 +5195,25 @@ packages:
     engines: {node: '>=16'}
     dev: true
 
+  /@edge-runtime/primitives@3.1.1:
+    resolution: {integrity: sha512-ROO22py+KdAfzqWZu6CtVMC4qV6mS0W1jPI51jGXE+uenyBUN7cQTWB9ReQc8Bm4cnjqmhajvpqEx3j7Y9iSOg==}
+    engines: {node: '>=16'}
+    dev: true
+
   /@edge-runtime/vm@3.1.0:
```
Comment (styfle) [inline pnpm-lock.yaml:5203]: Is it expect that both `vm@3.1.0` and `vm@3.1.1` are used?

**Review by styfle — APPROVED** [review]

File: pnpm-lock.yaml (hunk 1 above)
Comment (Kikobeats) [reply pnpm-lock.yaml:5203]: it should be the same 🤔

**Review by huozhi — APPROVED** [review]

**Comment by Kikobeats** [comment]
Recreated at https://github.com/vercel/next.js/pull/55062

**Pull request closed without merging**